astropy/utils/data.py

```diff
@@ -1174,25 +1212,50 @@ def download_files_in_parallel(urls, cache=True, show_progress=True,
         Timeout for each individual requests in seconds (default is the
         configurable `astropy.utils.data.Conf.remote_timeout`).
 
+    sources : dict, optional
+        If provided, for each URL a list of URLs to try to obtain the
+        file from. The result will be stored under the original URL.
+        For any URL in this dictionary, the original URL will *not* be
+        tried unless it is in this list; this is to prevent long waits
+        for a primary server that is known to be inaccessible at the
+        moment.
+
+    multiprocessing_start_method : str, optional
+        Useful primarily for testing; if in doubt leave it as the default.
+        When using multiprocessing, certain anomalies occur when starting
+        processes with the "spawn" method (the only option on Windows);
+        other anomalies occur with the "fork" method (the default on Unix).
+
     Returns
     -------
     paths : list of str
         The local file paths corresponding to the downloaded URLs.
+
+    Note
+    ----
+    If a URL is unreachable, the downloading will grind to a halt and the
+    exception will propagate upward, but an unpredictable number of
+    files will have been successfully downloaded and will remain in
+    the cache.
```

Comment:
This seems like the right behaviour. If the files can't all be downloaded, the user gets an exception as soon as possible. Presumably they wanted them all. Any where the download had actually started (I think) will be saved in the cache. The rest still need to be downloaded (or all of them if the user wants cache="update"). But there's no easy way for a user to call this to say "stuff the cache with everything you can".